### DISCUSSION TITLE
Internet radio shows title only when stopped

### DISCUSSION THREAD

**Discussion by jelinj8** [rern/rAudio#104] [Fixed Issues]
I've added an internet radio stream to my rAudio on Pi with a 16x2 LCD.
```
ČRo D-Dur
https://ice7.radia.cz/croddur256.mp3
```
When the player is stopped, it shows as `ČRo Radiožurnál` (without accents on the LCD), but when playing it shows just three dots (flashing on web GUI, stable on the LCD).

The stream probably doesn't provide expected metadata, but I think that we should keep the defined title in that case.

**Comment by ceejayemm** [comment]
@rern 
I will take this one stage further. With my RPi Zero W, RAudio-1 and update 20200113 applied, accessed from my PC with MS Edge, if I choose a Webradio station from the Library and select 'Replace' then if the station has a logo applied, eg Radio Paradise - Main Mix, then the existing Playlist is replaced. However if the webradio doesn't have a logo attached then the station is ADDED to the playlist and does not replace any existing playlist. The Playlist shows the webradio url but not the station name shown in the Library and like @jelinj8 above the name is missing from the main screen and the VU Meter (not the selected logo)  is active. 

If I apply a logo to the station which uses the default VU meter then I am allowed to select the required logo and apply it but it doesn't seem to get applied/saved and subsequently is added to the playlist as above. My logos are a mixture of .jpg and .png files and have worked correctly until this update.

Chris

**Comment by ceejayemm** [comment]
@rern 
It gets worse. Radio stations can't be deleted from the Library. If an attempt is made to delete a station then it momentarily disappears from the list but then reappears. If you don't notice this and a new station is added with the same details as the deleted station then a duplicate station listing is created.

If it helps the Webradio Library list in question was created a little while ago when testing the import processes for radio stations. All imported stations appear to work correctly in that they sound OK.

Chris

**Comment by ceejayemm** [comment]
@rern 
With Update 20200115 I can see that you are trying to do something to fix the web radios issues. However the update fails with:

/srv/http/settings/system.php
cat: 
'/srv/http/data/webradios/http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_radio_two|bbc_radio_two.isml|bbc_radio_two-audio%3d320000.norewind.m3u8'
: No such file or directory

If I login via SSH and go to the 'srv/http/data/webradios' folder, I can list the files:

rAudio:/srv/http/data/webradios  #  ls -lrt
total 104
-rwxr-xr-x 1 http http 55 Aug 18 03:49 'http:||stream.radioparadise.com|world-etc-flacm'
-rwxr-xr-x 1 http http 54 Aug 18 03:49 'http:||stream.radioparadise.com|rock-flacm'
-rwxr-xr-x 1 http http 56 Aug 18 03:49 'http:||stream.radioparadise.com|mellow-flacm'
-rwxr-xr-x 1 http http 54 Aug 18 03:49 'http:||stream.radioparadise.com|flacm'
-rw-r--r-- 1 http http 12 Jan  3 10:40 'http:||str1.sad.ukrd.com:80|eagle'$'\r'
-rw-r--r-- 1 http http 17 Jan  3 10:40 'http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_radio_five_live|bbc_radio_five_live.isml|bbc_radio_five_live-audio%3d320000.norewind.m3u8'$'\r'
-rw-r--r-- 1 http http  5 Jan  3 10:40 'http:||media-ice.musicradio.com:80|Gold'$'\r'
-rw-r--r-- 1 http http 12 Jan  3 10:40 'http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_radio_two|bbc_radio_two.isml|bbc_radio_two-audio%3d320000.norewind.m3u8'$'\r'
-rw-r--r-- 1 http http 15 Jan  3 10:40 'http:||88.208.208.140:8802|stream'$'\r'
-rw-r--r-- 1 http http 12 Jan  3 10:40 'http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_6music|bbc_6music.isml|bbc_6music-audio%3d320000.norewind.m3u8'$'\r'
-rw-r--r-- 1 http http 28 Jan  3 10:40 'http:||stream.radioparadise.com|mellow-flacm'$'\r'
-rw-r--r-- 1 http http 20 Jan  3 10:40 'https:||stream-mz.planetradio.co.uk|net2surrey.aac'$'\r'
-rw-r--r-- 1 http http 26 Jan  3 10:40 'http:||stream.radioparadise.com|flacm'$'\r'
-rw-r--r-- 1 http http 20 Jan  3 10:40 'http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_radio_berkshire|bbc_radio_berkshire.isml|bbc_radio_berkshire-audio%3d320000.norewind.m3u8'$'\r'
-rw-r--r-- 1 http http 26 Jan  3 10:40 'http:||stream.radioparadise.com|rock-flacm'$'\r'
-rw-r--r-- 1 http http 17 Jan  3 10:40 'http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_radio_surrey|bbc_radio_surrey.isml|bbc_radio_surrey-audio%3d320000.norewind.m3u8'$'\r'
-rw-r--r-- 1 http http 11 Jan  3 10:40 'http:||5.9.11.67:8208|stream.mp3'$'\r'
-rw-r--r-- 1 http http 25 Jan  3 10:40 'http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_radio_five_live_sports_extra|bbc_radio_five_live_sports_extra.isml|bbc_radio_five_live_sports_extra-audio%3d320000.norewind.m3u8'$'\r'
-rw-r--r-- 1 http http 13 Jan  3 10:40 'http:||radio.virginradio.co.uk|stream'$'\r'
-rw-r--r-- 1 http http 17 Jan  3 10:40 'http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_radio_solent|bbc_radio_solent.isml|bbc_radio_solent-audio%3d320000.norewind.m3u8'$'\r'
-rw-r--r-- 1 http http 12 Jan  3 10:40 'https:||stream-mz.planetradio.co.uk|scalahigh.aac'$'\r'
-rw-r--r-- 1 http http 27 Jan  3 10:40 'http:||stream.radioparadise.com|eclectic-flacm'$'\r'
-rw-r--r-- 1 http http 13 Jan  3 10:40 'http:||str1.sad.ukrd.com:80|pirate'$'\r'
-rw-r--r-- 1 http http 12 Jan 14 09:58 'http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_6music|bbc_6music.isml|bbc_6music-audio=320000.norewind.m3u8'
-rw-r--r-- 1 http http 28 Jan 14 18:00 'http:||str1.sad.ukrd.com:80|pirate'
-rw-r--r-- 1 http http 27 Jan 14 18:01 'https:||stream-mz.planetradio.co.uk|net2surrey.aac'
rAudio:/srv/http/data/webradios  #

You can see that some, not all files have an extra '$'\r' appended to the end of the file name. Not sure where this came from (may be from #53 importing radio station .pls files ?). The file in the Update 'cat' command ('/srv/http/data/webradios/http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_radio_two|bbc_radio_two.isml|bbc_radio_two-audio%3d320000.norewind.m3u8') only exists as:

'/srv/http/data/webradios/http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_radio_two|bbc_radio_two.isml|bbc_radio_two-audio%3d320000.norewind.m3u8'$'\r'

There are also some 'duplicate' files, one with the extra characters and one without:

rAudio:/srv/http/data/webradios  #  ls -lrt 'http:||str1.sad.ukrd.com:80|pirate'*
-rw-r--r-- 1 http http 13 Jan  3 10:40 'http:||str1.sad.ukrd.com:80|pirate'$'\r'
-rw-r--r-- 1 http http 28 Jan 14 18:00 'http:||str1.sad.ukrd.com:80|pirate'

The first file (with the extra chars) contains:
Pirate Radio

Whilst the second file contains:
(blank line)
16 bit 44.1 kHz 128 kbit/s

Chris

**Reply by rern** [reply]
Your `*.pls` files contain `\r` for "end of line" instead of `\n`.

End of line:
- `\n` Linux, OS X
- `\r\n` Windows
- `\r` Macs OS 9 and older

Use only text editors that can set it properly.

**Comment by ceejayemm** [comment]
@rern 
I have converted all my .pls files to unix/linux format, deleted my existing files in /webradios and /webradiosimg and then reimported the .pls files. This seems to have solved the problems above  but I still have a problems with the logos for some stations.  For example, the BBC Radio Two url:

http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio%3d320000.norewind.m3u8

when used with the default VU Meter is displayed correctly on all screens but when I apply the R2 logo in its place the R2 logon is shown correctly in the Webradio list and in the Playlist screens but is not displayed at all on the main Playback screen.

However the Eagle Radio url:

http://str1.sad.ukrd.com:80/eagle

displays the default VU meter correctly in all screens and also displays the replacement Eagle radio logo correctly in all areas too.

Chris

**Comment by ceejayemm** [comment]
@rern 

Still the same after applying update 20210120 (RPi Zero W, Bluethooth speakers only)

Chris

**Comment by ceejayemm** [comment]
@rern 

Fixed it.

The BBC radio station urls were imported from .pls files (from http://www.radiofeeds.co.uk/) where the urls include '%3d320000' which rAudio keeps for the station url but translates to '=320000' for the thumbnail images. So, for example, I have changed the station url from:

http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio%3d320000.norewind.m3u8

to

http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio=320000.norewind.m3u8

and the station logos are now displayed correctly. May be rAudio should be a bit more consistent in url/image naming ? If it translates for the image may be it should do the same to the url (or vice versa) ?

Chris

**Reply by rern** [reply]
The `%3d` for `=` should not be there in the first place.
